src/e3/anod/sandbox/__init__.py

```diff
@@ -40,21 +39,45 @@ def __init__(self) -> None:
             "vcs",
             "patch",
         )
-
-        self.meta_dir: Optional[str] = None
-        self.tmp_dir: Optional[str] = None
-        self.tmp_cache_dir: Optional[str] = None
-        self.src_dir: Optional[str] = None
-        self.log_dir: Optional[str] = None
-        self.vcs_dir: Optional[str] = None
-        self.patch_dir: Optional[str] = None
-        self.bin_dir: Optional[str] = None
-        self.__specs_dir: Optional[str] = None
+        self.meta_dir: str = ""
```

Comment:
Just a question - not a request for change, especially since this would be another change independent from what you are trying to do, here:

Would it make sense, eventually, to pre-initialize those to default directories inside the given root_dir (now that root_dir is going to be mandatory)? One interesting upside of this, I think, is that it would perhaps allow us to get rid of the "dirs" attribute (assuming no one outside the class started using it)?